pkg/controllers/cluster/status_controller.go

```diff
@@ -72,7 +75,16 @@ func (r *ClusterStatusReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	// patch message and condition
 	result, err := clientutil.PatchStatus(ctx, r.Client, cluster, func() error {
 		cluster.Status.KubernetesVersion = k8sVersion
-		cluster.Status.SetConditions(readyCondition, allNodesReadyCondition, kubeConfigValidCondition)
+		conditions = append(conditions, readyCondition, allNodesReadyCondition, kubeConfigValidCondition)
```

Comment:
Can we move the computation of the `ClusterDeletionScheduled` into a dedicated function and out of the `PatchStatus`?

Reply:
done https://github.com/cloudoperators/greenhouse/pull/609/commits/713a589db360729839dd105c2608d93dab348792

validated with a test